determined-ai/determined
# fix: Workspace store quickly reflects add/remove [WEB-635]

## Description

For the list of pinned workspaces to update immediately after creating or deleting a workspace, we need to update the local workspaces store after create/delete API calls succeed.

In the current PR, the true API calls move to the store `stores/workspaces`, and the modal's line `await createWorkspace(body);` now directs the store to handle the API call and updating itself.

Note: Pinned workspaces are ordered by however the API responds (currently old -> new). For example if I wrote `[w].concat(ws)` it would temporarily add the new workspace at the front of the pinned workspaces array. I rewrote it for consistency then added a `sort()` to the pinned workspaces view to enforce a consistent order.

## Test Plan

- In the left sidebar, scroll down to the last pinned workspace (new workspaces add to the bottom of the list)
- Click the nearby [+] button to open the Create Workspace Modal. Give a name such as "test"
- After clicking OK, the modal disappears and workspace should get simultaneously added on the list.
- Scroll the pinned workspaces again so the new workspace is visible. Delete it and it should simultaneously be removed from the pinned workspaces list

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
